### PR TITLE
Update core/contrib deps to v0.38.0

### PR DESCRIPTION
**Pull request by jcheng-splunk**
Integration tests passed locally.

### CODE REVIEW

File: internal/buildscripts/update-deps [hunk 1]
```diff
@@ -34,7 +34,7 @@ for gomod in $( find "$REPO_DIR" -name "go.mod" | grep -v "/examples/" | sort );
     fi
 
     set +o pipefail
-    lines="$( grep 'github.com/open-telemetry/opentelemetry-collector-contrib/' "$gomod" | grep -v '=>' | grep -v ' // indirect' | sort -u )"
+    lines="$( grep 'github.com/open-telemetry/opentelemetry-collector-contrib/' "$gomod" | grep -v '^[[:space:]]*//' | grep -v '=>' | grep -v ' // indirect' | sort -u )"
```
Comment (dmitryax) [inline internal/buildscripts/update-deps:37]: Are these couple changes related to the PR?

Comment (jcheng-splunk) [reply internal/buildscripts/update-deps:37]: I use this script for the dep updates and needed to ignore the comments.

**Review by rmfitzpatrick — APPROVED** [review]
